openapi.yaml

```diff
@@ -0,0 +1,513 @@
+                  gitRepo: https://github.com/bigcorp/commodore-config.git
+                - id: os3ce3
+                  name: acmecorp
+                  displayName: Acme Corp. (Subtenant of Big Corp)
+                  tenant: aezoo6
+                  gitRepo: https://github.com/acmecorp/commodore-config.git
+        default:
+          $ref: '#/components/responses/Default'
+    post:
+      summary: Creates a new tenant
+      description: |
+        Create a tenant in the API.
+        * ID is generated by the API (6 lowercase letters and numbers mixed)
+        * Generates the `Tenant` object in the same namespace as the API runs
+        * The customer config Git repository (the `<GitRepoSpec>`) is automatically
+          generated based on default configuration or using the provided values
+        * If `gitRepo` is already set on creation, no `<GitRepoSpec>` will be added.
+      tags:
+        - tenant
+      requestBody:
+        required: true
+        $ref: '#/components/requestBodies/CreateTenant'
+      responses:
+        '201':
+          description: Tenant created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Tenant'
+              example:
+                id: aezoo6
+                name: acmecorp
+                displayName: Acme Corp.
+                tenant: multicorp
+                gitRepo: https://github.com/acmecorp/commodore-config.git
+        '400':
+          description: Tenant cannot be created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Reason'
+        '405':
+          description: Tenant already exists
+        default:
+          $ref: '#/components/responses/Default'
+  /tenants/{tenantId}:
+    get:
+      summary: Returns all values of a tenant
+      tags:
+        - tenant
+      parameters:
+        - $ref: '#/components/parameters/TenantId'
+      responses:
+        '200':
+          description: Tenant found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Tenant'
+        '404':
+          description: A tenant with the specified id was not found.
+        default:
+          $ref: '#/components/responses/Default'
+    patch:
+      summary: Updates a tenant
+      tags:
+        - tenant
+      parameters:
+        - $ref: '#/components/parameters/TenantId'
+      requestBody:
+        description: Update tenant with properties to be changed (RFC 7396)
+        required: true
+        content:
+          application/merge-patch+json:
+            schema:
+              allOf:
+                - $ref: '#/components/schemas/Tenant'
+                - type: object
+                  properties:
+                    displayName:
+                      nullable: true
+      responses:
+        '204':
+          description: Tenant updated
+        '403':
```

Comment:
Forbidden usually indicates an authorization issue. What about those cases where the user provided invalid input?

I have seen that  _422 Unprocessable Entity_ was used which is from the WebDAV.

Reply:
May I propose to improve on this in a future iteration?

Reply:
We'll work on iterating HTTP status codes later.